### PR TITLE
ENH PHP 8.1 compatibility

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10250

This PR is for Graphql 3

### CODE REVIEW

File: src/Controller.php [hunk 1]
```diff
@@ -515,14 +514,14 @@ public static function flush()
...
-            if (stristr($routeClass, Controller::class) !== false) {
+            if (stristr($routeClass ?? '', Controller::class) !== false) {
                 try {
                     $inst = Injector::inst()->convertServiceProperty($routeClass);
                     if ($inst instanceof Controller) {
                         /* @var Controller $inst */
                         $inst->processTypeCaching();
                     }
-                } catch (DatabaseException $e) {
+                } catch (Exception $e) {
```
Comment (GuySartorelli) [inline src/Controller.php:524]: Which exceptions are coming through that aren't `DatabaseException`? The comment in this catch says it's specifically there to catch failures where the table doesn't exist etc so it's specifically for database exceptions - I'm a bit worried about catching legitimate exceptions here.

**Review by GuySartorelli — COMMENTED** [review]
The outstanding question on other PRs about `file_get_contents()` will also affect this PR.

File: src/Controller.php (hunk 1 above)
Comment (emteknetnz) [reply src/Controller.php:524]: Oh this is wrong.  It was for the `mysql_sqi_exception` that was being triggered by php 8.1.  I must have not reverted, it should be DatabaseException.

**Review by GuySartorelli — APPROVED** [review]
For posterity: See discussion in silverstripe/silverstripe-admin#1294 about the general approach. tl;dr:
> the purposes of this PR is to avoid passing null (specifically null) from built-in functions that as of php 8.1 will throw deprecation warnings if null is passed in. The approach is necessarily heavy-handed, and while there are many situations where it isn't needed, it would be prohibitively laborious to find all of those situations by hand.

Looks good to me

**Pull request merged**